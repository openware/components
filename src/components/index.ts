export * from './button';
export * from './landing-block';
export * from './icon';
export * from './widget-frame';
export * from './tabs';
export * from './vtable';
export * from './trade-history';
