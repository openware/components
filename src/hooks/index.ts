export * from './use-did-mount';
export * from './redux';
export * from './use-outside-click';
