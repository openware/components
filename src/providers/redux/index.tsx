import * as React from 'react';
import { Provider } from 'react-redux';

// eslint-disable-next-line
const ReduxProvider = ({ children, store }) => (
    <Provider store={store}>{children}</Provider>
);

export { ReduxProvider };
