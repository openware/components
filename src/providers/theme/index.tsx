import React, {
    useContext,
    useState,
    useCallback,
    useEffect,
    createContext,
    useMemo,
} from 'react';
import classNames from 'classnames';
import { ThemeName } from './names';

export { ThemeName };

export interface ThemeContextProps {
    theme: ThemeName;
    rootClassName: string;
    changeTheme: (name: ThemeName) => void;
}

function getRootClassName(themeName: ThemeName): string {
    return classNames('app-root-theme', `app-root-theme--${themeName}`);
}

const DefaultTheme: ThemeName = 'light';

const ThemeContext = createContext<ThemeContextProps>({
    theme: DefaultTheme,
    rootClassName: getRootClassName(DefaultTheme),
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    changeTheme: () => {},
});

interface Props {
    theme?: ThemeName;
    children: React.ReactNode;
}

export const ThemeProvider: React.FC<Props> = ({ theme, children }) => {
    const [innerTheme, setInnerTheme] = useState<ThemeName>(DefaultTheme);

    const handleChange = useCallback((newTheme: ThemeName) => {
        setInnerTheme(newTheme);
    }, []);

    useEffect(() => {
        if (theme) {
            setInnerTheme(theme);
        }
    }, [theme]);

    const themeValue = useMemo<ThemeContextProps>(() => {
        return {
            theme: innerTheme,
            changeTheme: handleChange,
            rootClassName: getRootClassName(innerTheme),
        };
    }, [innerTheme, handleChange]);

    return (
        <ThemeContext.Provider value={themeValue}>
            {children}
        </ThemeContext.Provider>
    );
};

export function useTheme(): ThemeContextProps {
    return useContext(ThemeContext);
}
