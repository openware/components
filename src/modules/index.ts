import { combineReducers } from 'redux';
import { generalReducer, GeneralState } from './general';

export * from './general';

export const rootReducer = combineReducers({
    general: generalReducer,
});

export interface RootState {
    general: GeneralState;
}
