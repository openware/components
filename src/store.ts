import { createStore } from 'redux';
import { rootReducer } from 'src/modules';

export const store = createStore(rootReducer);
