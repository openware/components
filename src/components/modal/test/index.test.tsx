import * as React from 'react';
import { shallow } from 'enzyme';

import { Modal, ModalProps } from '../index';
import { ReduxProvider } from 'src/providers';
import { store } from 'src/store.ts';

const defaultProps: ModalProps = {
    name: '',
    children: <div />,
    onOk: jest.fn(),
};

const setup = (props: Partial<ModalProps> = {}) => {
    return shallow(
        <ReduxProvider store={store}>
            <Modal {...defaultProps} {...props} />
        </ReduxProvider>
    );
};

describe('Modal', () => {
    it('snapshot render', () => {
        expect(setup().render()).toMatchSnapshot();
    });

    it('should have correct parent className', () => {
        const component = setup().render();

        expect(component.hasClass('modal')).toBeTruthy();
    });

    it('should have additional className if defined', () => {
        const customClassName = 'modal__custom-class';
        const component = setup({ className: customClassName }).render();

        expect(component.hasClass(customClassName)).toBeTruthy();
        expect(component.hasClass('modal')).toBeTruthy();
    });
});
