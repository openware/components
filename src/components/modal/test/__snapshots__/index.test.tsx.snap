// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal snapshot render 1`] = `
<div
  class="modal"
>
  <div
    class="modal__container"
  >
    <div
      class="modal__container-header"
    />
    <div
      class="modal__container-content"
    >
      <div />
    </div>
    <div
      class="modal__container-footer"
    >
      <button
        class="test-button modal__container-footer-cancel"
      >
        Cancel
      </button>
      <button
        class="test-button modal__container-footer-confirm"
      >
        Confirm
      </button>
    </div>
  </div>
</div>
`;
